src/core/thermostat.hpp

```diff
@@ -92,6 +92,24 @@ extern bool thermo_virtual;
  ************************************************/
 
 struct BaseThermostat {
+public:
+  /** Initialize or re-initialize the RNG counter with a seed. */
+  void rng_initialize(uint64_t const seed) {
+    rng_counter = std::make_unique<Utils::Counter<uint64_t>>(seed);
+  }
+  /** Increment the RNG counter */
+  void rng_increment() { rng_counter->increment(); }
+  /** Get current value of the RNG */
+  uint64_t rng_get() const {
+    if (!rng_is_initialized()) {
+      throw "The RNG is not initialized";
+    }
+    return rng_counter->value();
+  }
+  /** Is the RNG counter initialized */
+  bool rng_is_initialized() const { return rng_counter != nullptr; }
+
+private:
   /** RNG counter. */
   std::unique_ptr<Utils::Counter<uint64_t>> rng_counter;
```

Comment:
There is no need anymore to store this on the heap right? A `boost::optional<uint64_t>` would too I think.